### PR TITLE
perf(desktop): cut sidebar hover and polling render fan-out

### PR DESCRIPTION
**Pull request by Kitenite**
Stacked follow-up to #6364 (sidebar pin drag-and-drop). Addresses the three pre-existing sidebar rendering costs identified while profiling that work.

## Summary
- **Hover-open PR refresh gets a cool-down** (`pullRequestRefreshCooldown`, unit-tested): opening a row's hover card no longer fires an unthrottled host `pullRequests.refreshByWorkspaces` mutation + query invalidation per hover.
- **~80 per-row query subscriptions collapse into one**: new `DashboardSidebarWorkspaceStatusProvider` holds a single sidebar-level subscription with per-row selectors (a row re-renders only when *its* entry changes); the hover provider is split so rows consume a stable actions context instead of churning state.
- **Per-row framer-motion wrappers replaced with CSS row-collapse**: only the whole-project expand/collapse motion wrapper remains. Section collapse still animates; rows still hide while their section header is dragged.

## Why / Context
CPU profiles during the #6364 work showed a plain mouse sweep across sidebar rows costing ~358ms of long tasks (worst 129ms) with `shallowEqualObjects`/`useBaseQuery` hot — hover-open side effects, per-row react-query subscriptions notified on every 10s PR poll, and ~80 motion components.

## How It Works
- The cool-down is a module-level per-workspace timestamp map checked before firing the refresh; semantics of *what* refreshes are unchanged, only the rate.
- `DashboardSidebarWorkspaceStatusProvider` subscribes once and exposes selector hooks; poll ticks no longer run ~80 per-row subscription compare passes.
- The hover provider now exports a state context and a separate stable actions context; the DnD provider and rows consume actions without re-rendering on hover-state changes.

## Verification
Authored by an orchestrated worker agent; independently re-verified by the coordinator, then rebased onto post-#6364 main with one conflict reconciled (`DashboardSidebarDndProvider`: kept the actions-context split *and* the try/finally suppression release + fire-time open-timer re-check from 1aab78f3f — both suppression guards confirmed present after the merge).

- `bun run typecheck`, `bun run lint` (exit 0, no post-lint diff), 92/92 tests across the sidebar + state + host-service-hook suites — run on the rebased branch.
- CDP on the rebased branch (worker worktree stack, correct-instance verified): pin-by-drag works, and the drag-in-then-drag-out single gesture (the #6364 review finding) completes both transfers; zero React errors.
- Hover-sweep long tasks, same methodology as the baseline: **358ms / 4 tasks / 129ms worst → 178ms / 2 tasks / 90ms worst** in my measurement window (the worker measured 0ms in a quieter window; the residual in mine coincides with the 10s PR-poll tick, which this PR reduces in fan-out but deliberately does not re-schedule).

## Manual QA for reviewer (dev dataset couldn't exercise these)
- [ ] Agent status dots on sidebar rows render/update correctly with live running agents (status data now flows through the new provider)
- [ ] Section group collapse/expand animation looks right with real section groups (CSS transition replaced per-row framer-motion)

## Risks / Rollback
- Risk: status/notification display regressions (every row's status now flows through one provider). Mitigated by preserved query semantics + suites; the two QA boxes above are the uncovered surface.
- Rollback: revert this single commit; no schema or persisted-state changes.

---
## Summary by cubic
Reduce sidebar render fan-out on hover and polling to cut long-task time during a hover sweep by ~50% and limit re-renders to only affected rows.

- **Performance**
  - Add a 45s cool-down for hover-open PR refreshes to avoid repeated `pullRequests.refreshByWorkspaces` calls per hover (unit-tested).
  - Introduce `DashboardSidebarWorkspaceStatusProvider` to replace ~80 per-row query subscriptions with one sidebar-level subscription + per-row selectors; also centralizes lifecycle event listeners and diff-stats invalidation.
  - Split hover provider into a stable actions context plus a per-row `isHovered` subscription so a hover commit re-renders only the two involved rows.
  - Replace per-row framer-motion wrappers with a CSS grid-row/opacity transition; rows stay mounted but inert; section animations preserved.

<sup>Written for commit 99e71a6b3908c1fb28c074a46bae7313254a2e0f. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/superset-sh/superset/pull/6377?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

## Summary by CodeRabbit

* **New Features**
  * Added consolidated workspace status indicators to the dashboard sidebar, including attention states, terminal activity, and diff statistics.
  * Improved sidebar hover behavior and transitions for smoother interactions.
  * Added workspace-specific unread and attention counts.
* **Performance**
  * Limited pull-request refreshes to once every 45 seconds per workspace.
* **Bug Fixes**
  * Improved sidebar status updates and terminal-seen handling across workspace rows.
* **Tests**
  * Added coverage for pull-request refresh cooldown behavior.

### CODE REVIEW

**Comment by Kitenite** [comment]
## CDP stress battery on this branch (same methodology as #6364's)

All phases driven with trusted `Input.dispatchMouseEvent` on this branch's dev app, with persisted-state assertions and a console-error hook. Video: 25s reel attached to the review thread offline (section collapse, hover+drag suppression, header drag, pin roundtrip).

**Verified:**
- **Hover-open cool-down works**: 12 hover-card opens across 4 rows produced exactly **4 `refreshByWorkspaces` mutations** (one per distinct workspace; repeat hovers suppressed).
- **No suppression leaks**: 8 rapid hover→drag→drop→hover cycles — 0 cards mid-drag, hover cards still open normally afterwards; try/finally paths hold.
- **Section CSS collapse works** (live-created section group): clip-aware probing (ancestor `overflow: hidden` wrapper collapses to 0 height, `elementFromPoint` confirms the next project moves up) — expand/collapse storm of 14 rapid toggles, clean. Note for anyone testing this later: `getBoundingClientRect()` on the rows is blind to the CSS clipping — probe via ancestor height or hit-testing.
- **Pin roundtrip with sections**: section member → Pinned (sectionId retained on the row for return) → back into the section, all persisted correctly. One observed boundary case is pre-existing #6364 semantics, not this PR: a drop landing exactly at a section header boundary assigns membership by visual position (my test hit it with stale mid-drag coordinates).
- **Status-dot churn** (20 cycles × 10-workspace unread toggles through the new provider): steady-state toggles are **0ms long tasks — identical to the pre-PR baseline** (A/B measured). The *first* toggle for a given row can cost a few hundred ms (query warm-up through the provider); bounded, non-recurring, and rare in real use, but worth knowing.
- **Zero "Maximum update depth" / render-loop errors** across every phase. The one console error observed is the pre-existing `DockBadgeController` setState-in-render warning (present on main; the new provider merely changed where it fires from — could be a tiny follow-up).

File: apps/desktop/src/renderer/routes/_authenticated/_dashboard/components/DashboardSidebar/providers/DashboardSidebarWorkspaceStatusProvider/DashboardSidebarWorkspaceStatusProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+					binding.terminalId,
+					deriveTerminalAgentStatus({
+						lastEventType: binding.lastEventType,
+						lastEventAt: binding.lastEventAt,
+						lastSeenAt: terminalSeenAt[binding.terminalId],
+					}),
+				);
+			}
+			const hasManualUnread = Boolean(manualUnread[target.workspaceId]);
+			let hasAttentionTerminal = false;
```
[the remark this answers is not in the document]
Comment (Kitenite) [reply apps/desktop/src/renderer/routes/_authenticated/_dashboard/components/DashboardSidebar/providers/DashboardSidebarWorkspaceStatusProvider/DashboardSidebarWorkspaceStatusProvider.tsx:288]: Not a regression — the pre-PR row-level `useV2WorkspaceIsUnread` used exactly `review || failed` (see bb3daa48d, useV2NotificationStatus.ts:84); the provider preserves it. `permission` was only ever counted by `useV2AttentionWorkspaceCount` (dock badge), which is intentionally broader — permission shows as the red status dot, not the unread mark. Pre-existing split between two metrics; happy to align them in a follow-up if we want them unified.

File: apps/desktop/src/renderer/routes/_authenticated/_dashboard/components/DashboardSidebar/components/DashboardSidebarProjectSection/components/DashboardSidebarExpandedProjectContent/DashboardSidebarExpandedProjectContent.tsx [hunk 2]
```diff
@@ -132,49 +133,54 @@ export function DashboardSidebarExpandedProjectContent({
...
+															: undefined
+													}
+													disabled={
+														hidden ||
+														(workspace.type === "main" &&
+															workspace.hostType === "local-device")
+													}
+												/>
+											</div>
+										</div>
```
[the remark this answers is not in the document]
Comment (Kitenite) [reply apps/desktop/src/renderer/routes/_authenticated/_dashboard/components/DashboardSidebar/components/DashboardSidebarProjectSection/components/DashboardSidebarExpandedProjectContent/DashboardSidebarExpandedProjectContent.tsx:183]: Real trade-off, acknowledged: collapsed-section rows now stay mounted (clipped) instead of unmounting, and the provider computes entries for them. Each mounted row is far cheaper than before (memoized body + per-row selector), and measured sweeps improved net, but large collapsed sections do pay a new baseline. Follow-up candidate: skip status-entry computation for rows inside collapsed sections, or render clipped rows as placeholders. Deferring rather than blocking — the current numbers are strictly better than main for typical sidebars.

File: apps/desktop/src/renderer/routes/_authenticated/_dashboard/components/DashboardSidebar/components/DashboardSidebarWorkspaceItem/DashboardSidebarWorkspaceItem.tsx [hunk 3]
```diff
@@ -136,10 +133,12 @@ export function DashboardSidebarWorkspaceItem({
...
 
-	const isHovered = hoverHoveredId === id;
+	const isHovered = useDashboardSidebarIsHovered(id);
 	useEffect(() => {
-		if (isHovered && hostType === "local-device") onHoverCardOpen?.();
-	}, [isHovered, hostType, onHoverCardOpen]);
+		// Fires on the committed hover only (hoveredId set after OPEN_DELAY or an
+		// open-card switch), never on transient row mouseenter.
+		if (isHovered && hostType === "local-device") void onHoverCardOpen?.(id);
+	}, [isHovered, hostType, onHoverCardOpen, id]);
```
[the remark this answers is not in the document]
Comment (Kitenite) [reply apps/desktop/src/renderer/routes/_authenticated/_dashboard/components/DashboardSidebar/components/DashboardSidebarWorkspaceItem/DashboardSidebarWorkspaceItem.tsx:141]: Valid — the hover-open handler is now async and the call site discards the promise; a host-offline rejection would surface as an unhandled rejection. Queued as a follow-up quick-win batch together with the cooldown-on-failure and map-prune notes below.

File: apps/desktop/src/renderer/routes/_authenticated/_dashboard/components/DashboardSidebar/hooks/useDashboardSidebarData/useDashboardSidebarData.ts [hunk 4]
```diff
@@ -444,6 +448,9 @@ export function useDashboardSidebarData() {
 				(candidate) => candidate.machineId === workspace.hostId,
 			);
 			if (!target?.hostUrl) return;
+			if (!pullRequestRefreshGate.shouldRefresh(workspaceId, Date.now())) {
```
[the remark this answers is not in the document]
Comment (Kitenite) [reply apps/desktop/src/renderer/routes/_authenticated/_dashboard/components/DashboardSidebar/hooks/useDashboardSidebarData/useDashboardSidebarData.ts:451]: Valid but low-impact by design: the 10s PR poll is the freshness backstop, so a failed hover-refresh being suppressed for one 45s window only delays the *hover-triggered* fast path, not correctness. Will move the timestamp write to post-success in the follow-up batch.

File: apps/desktop/src/renderer/routes/_authenticated/_dashboard/components/DashboardSidebar/components/SortableWorkspaceItem/SortableWorkspaceItem.tsx [hunk 5]
```diff
@@ -10,7 +10,7 @@ interface SortableWorkspaceItemProps {
 	workspace: DashboardSidebarWorkspace;
 	accentColor?: string | null;
 	isInSection?: boolean;
-	onHoverCardOpen?: () => void;
+	onHoverCardOpen?: (workspaceId: string) => void | Promise<void>;
```
[the remark this answers is not in the document]
Comment (Kitenite) [reply apps/desktop/src/renderer/routes/_authenticated/_dashboard/components/DashboardSidebar/components/SortableWorkspaceItem/SortableWorkspaceItem.tsx:13]: Same as CodeRabbit's note on the item — folding into the follow-up quick-win batch (catch on the discarded promise).

File: apps/desktop/src/renderer/routes/_authenticated/_dashboard/components/DashboardSidebar/providers/DashboardSidebarWorkspaceStatusProvider/DashboardSidebarWorkspaceStatusProvider.tsx [hunk 6]
```diff
@@ -0,0 +1,382 @@
...
+				void queryClient.invalidateQueries({
+					queryKey: getTerminalAgentBindingsQueryKey(workspaceId),
+				});
+			};
+			cleanups.push(bus.on("agent:lifecycle", workspaceId, invalidateBindings));
+			cleanups.push(
+				bus.on("terminal:lifecycle", workspaceId, invalidateBindings),
+			);
+			cleanups.push(
+				bus.on("git:changed", workspaceId, () => {
```
[the remark this answers is not in the document]
Comment (Kitenite) [reply apps/desktop/src/renderer/routes/_authenticated/_dashboard/components/DashboardSidebar/providers/DashboardSidebarWorkspaceStatusProvider/DashboardSidebarWorkspaceStatusProvider.tsx:244]: Valid micro-dedup — will skip the active workspace in the sidebar-wide invalidation listener in the follow-up batch.

File: apps/desktop/src/renderer/routes/_authenticated/_dashboard/components/DashboardSidebar/hooks/useDashboardSidebarData/pullRequestRefreshCooldown.ts [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+	 * A `true` result records `now` as the workspace's last refresh, so
+	 * concurrent callers inside the window are rejected too.
+	 */
+	shouldRefresh: (workspaceId: string, now: number) => boolean;
+}
+
+export function createPullRequestRefreshGate(
+	cooldownMs: number = PULL_REQUEST_REFRESH_COOLDOWN_MS,
+): PullRequestRefreshGate {
+	const lastRefreshAtByWorkspaceId = new Map<string, number>();
```
[the remark this answers is not in the document]
Comment (Kitenite) [reply apps/desktop/src/renderer/routes/_authenticated/_dashboard/components/DashboardSidebar/hooks/useDashboardSidebarData/pullRequestRefreshCooldown.ts:20]: Valid — the map is tiny (8-byte timestamp per distinct hovered workspace per session) but pruning expired entries is a 3-liner; follow-up batch.
